### PR TITLE
chore: improving the compatibility checker

### PR DESCRIPTION
**Pull request by jskeet**
This commit is just to check that:

- The general approach makes sense
- The test data so far makes sense

### CODE REVIEW

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            {
+                { "1.1.0", Level.Major },        // So we know what we've broken compared with last GA
+                { "2.0.0-beta01", Level.Major }, // So we know what we've broken compared with last beta
+            }
+        };
+
+        // Non-release PRs will have a version that is already tagged (and published)
+        public static TheoryData<TestCase> NonReleasePrTestData = new TheoryData<TestCase>
+        {
+            // TODO
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:103]: Not sure if we could add a hint as to what the intended version is or if the code can run assuming the same version as increment-version would generate. We can always use allow breaking changes to bump major, but we'll have to continue to use it until we have released the first major beta.

**Review by amanda-tarafa — COMMENTED** [review]
Yep this looks good.

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            // First beta of a new major
+            new TestCase("2.0.0-beta01", "1.0.0", "1.1.0") // Any beta releases in 1.x are irrelevant now
+            {
+                { "1.1.0", Level.Major }
+            },
+
+            // Second beta of a new major
+            new TestCase("2.0.0-beta02", "1.0.0-beta01", "1.0.0", "1.1.0-beta01", "2.0.0-beta01")
+            {
+                { "1.1.0", Level.Major },        // So we know what we've broken compared with last GA
```
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:88]: The latter - done.

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                { "2.0.0-beta01", Level.Major }, // So we know what we've broken compared with last beta
+            },
+
+            // First release of a new major
+            new TestCase("2.0.0", "1.0.0-beta01", "1.0.0", "1.1.0-beta01", "2.0.0-beta01")
```
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:93]: Good call - done.

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:103]: I think if we know we'll do a major next, we can go to 2.0.0-beta00. Which reminds me, I don't mention that anywhere here...

Comment (amanda-tarafa) [reply tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:103]: Ah yes makes sense.

**Comment by jskeet** [comment]
This is now ready for review.

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            },
+
+            // New patch (no prerelease)
+            new TestCase("1.1.1", "1.0.0", "1.1.0")
+            {
+                { "1.1.0", Level.Identical }
+            },
+
+            // New patch (with prerelease)
+            new TestCase("1.1.1-beta01", "1.0.0", "1.1.0")
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:112]: Can you add one with 2 patch betas so we can break on the second with respect to the first?

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs (hunk 4 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs:112]: Yup! Adding now.

**Pull request merged**